code/UI/Player/PossessionNameplate/PossessionNameplate.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Sandbox;
+using Sandbox.UI;
+
+namespace TTT.UI;
+
+[UseTemplate]
+public class PossessionNameplate : Panel
+{
+	private Label PlayerName { get; set; }
+	private Player _player;
+	private Prop _prop;
+
+	public PossessionNameplate( Player player, Prop prop )
+	{
+		_player = player;
+		_prop = prop;
+		PlayerName.SetText( player.SteamName );
+		Local.Hud.AddChild( this );
+	}
+
+	public override void Tick()
+	{
+		if ( !_player.IsValid() || !_prop.IsValid() )
+		{
+			Delete();
+			return;
+		}
+
+		var occluded = Trace.Ray( CurrentView.Position, _prop.WorldSpaceBounds.Center )
+			.WorldAndEntities()
+			.Ignore( _prop )
+			.Run()
+			.Hit;
+
+		if ( occluded )
+		{
+			Style.Opacity = 0f;
+			return;
+		}
+
+		var screenPos = _prop.WorldSpaceBounds.Center.ToScreen();
+
+		Style.Opacity = 1;
+		Style.Left = Length.Fraction( screenPos.x );
+		Style.Top = Length.Fraction( screenPos.y );
+	}
+}
```

Comment:
This should use `WorldPanels`. Look at `RolePlate` for reference.

Reply:
> This should use WorldPanels. Look at RolePlate for reference.

One problem with `WorldPanel`s is that the nameplate is occluded by its own prop like in the following picture:

![NVIDIA_Share_MlwwBbUESv](https://user-images.githubusercontent.com/1663282/182870870-e7dc1af8-9754-4931-a188-a2275c485e2a.png)

Another problem is that the size of the nameplate changes the closer the camera gets to the prop, which does not happen in the original TTT.
Im not sure if these problems can be avoided if the nameplate is located at the prop's center position.

One way to work around this problem is to move the nameplate a bit closer to the camera which sort of works (nameplate can still be partially occluded by props and the size of the nameplate still changes a bit) as you can see here:

https://streamable.com/xnwbqq

Im probably just not thinking of the correct way to do this, but I think using normal `Panel`s looks at least better than what I came up with when using `WorldPanel`s.

Reply:
> Another problem is that the size of the nameplate changes the closer the camera gets to the prop, which does not happen in the original TTT. 

I think this is fine. We are using world panels for role plates and I think it makes to most sense to use it for prop possesion as well.

> One problem with WorldPanels is that the nameplate is occluded by its own prop like in the following picture:

Role plates have this same problem. S&box API let's you draw the object on the viewmodel layer so it can be drawn over the world, but this is currently broken. We will probably end up making an issue.